src/openrct2/management/Research.h

```diff
@@ -19,32 +19,72 @@ struct rct_ride_entry;
 
 struct ResearchItem
 {
-    // Bit 16 (0: scenery entry, 1: ride entry)
     union
     {
         uint32_t rawValue;
         struct
         {
-            uint8_t entryIndex;
+            ObjectEntryIndex entryIndex;
             uint8_t baseRideType;
             uint8_t type; // 0: scenery entry, 1: ride entry
-            uint8_t flags;
         };
     };
+    uint8_t flags;
     uint8_t category;
 
     bool IsNull() const;
+    void SetNull();
     bool Equals(const ResearchItem* otherItem) const;
```

Comment:
Is now a good time to replace `IsNull`, `SetNull`, `Equals` with operator overloads?

`if (item == ResearchItem::Null)` or `if (item == ResearchItem())` or `if (item == {})`
`item = {}` or `item = ResearchItem()`
`if (item == item2)`